web-admin/src/features/projects/status/ProjectResources.svelte

```diff
@@ -2,54 +2,92 @@
-  import RefreshConfirmDialog from "./RefreshConfirmDialog.svelte";
+  import RefreshAllSourcesAndModelsConfirmDialog from "./RefreshAllSourcesAndModelsConfirmDialog.svelte";
+  import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
 
   const queryClient = useQueryClient();
   const createTrigger = createRuntimeServiceCreateTrigger();
 
+  let isConfirmDialogOpen = false;
   let isReconciling = false;
-  let isRefreshConfirmDialogOpen = false;
+
+  const INITIAL_REFETCH_INTERVAL = 500; // Start at 500ms
+  const MAX_REFETCH_INTERVAL = 10_000; // Cap at 10s
+  const BACKOFF_FACTOR = 2; // Double each time
+  let currentRefetchInterval = INITIAL_REFETCH_INTERVAL;
 
   $: ({ instanceId } = $runtime);
 
-  $: resources = createRuntimeServiceListResources(
-    instanceId,
-    // All resource "kinds"
-    undefined,
-    {
-      query: {
-        select: (data) => {
-          // Filter out the "ProjectParser" resource
-          return data.resources.filter(
-            (resource) =>
-              resource.meta.name.kind !== "rill.runtime.v1.ProjectParser",
-          );
-        },
-        refetchOnMount: true,
-        refetchOnWindowFocus: true,
-        refetchInterval: isReconciling ? 500 : false,
-      },
-    },
-  );
+  function isResourceErrored(resource: V1Resource) {
+    return !!resource.meta.reconcileError;
+  }
 
-  $: isAnySourceOrModelReconciling = $resources?.data?.some(
-    (resource) =>
+  function isResourceReconciling(resource: V1Resource) {
+    return (
       resource.meta.reconcileStatus ===
         V1ReconcileStatus.RECONCILE_STATUS_PENDING ||
       resource.meta.reconcileStatus ===
-        V1ReconcileStatus.RECONCILE_STATUS_RUNNING,
+        V1ReconcileStatus.RECONCILE_STATUS_RUNNING
+    );
+  }
+
+  function calculateRefetchInterval(
+    currentInterval: number,
+    data: V1ListResourcesResponse | undefined,
+  ): number | false {
+    if (!data?.resources) {
+      return INITIAL_REFETCH_INTERVAL;
+    }
+
+    const hasErrors = data.resources.some(isResourceErrored);
+    const hasReconcilingResources = data.resources.some(isResourceReconciling);
+
+    if (hasErrors || !hasReconcilingResources) {
+      return false;
+    }
+
+    return Math.min(currentInterval * BACKOFF_FACTOR, MAX_REFETCH_INTERVAL);
+  }
+
+  $: resources = createRuntimeServiceListResources(instanceId, undefined, {
+    query: {
+      select: (data: V1ListResourcesResponse) => ({
+        ...data,
+        // Filter out project parser and refresh triggers
+        resources: data?.resources?.filter(
+          (resource: V1Resource) =>
+            resource.meta.name.kind !== ResourceKind.ProjectParser &&
+            resource.meta.name.kind !== ResourceKind.RefreshTrigger,
+        ),
+      }),
+      refetchInterval: (data) =>
+        calculateRefetchInterval(
+          $resources?.data ? currentRefetchInterval : INITIAL_REFETCH_INTERVAL,
```

Comment:
The `currentRefetchInterval` isn't getting updated, so this is currently polling every 1s w/o a backoff.